Build fails

Trying to build on [Artix GNU/Linux](http://artixlinux.org/), an Arch Linux based distribution, rolling release, so all packages are usually quite up to date with latest upstream release.

build according to this  [build recipe](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=gpxviewer-git):

```
meson --prefix /usr --buildtype=plain . builddir
ninja -C builddir
```

Build fails with :
```
gpx-viewer-path-layer.vala:259.3-259.31: error: overriding method `Gpx.Viewer.PathLayer.set_view' is incompatible with base method `void Champlain.Layer.set_view (Champlain.View?)': incompatible type of parameter 1.``:
```

The Meson build system

```
Version: 1.2.0
Source dir: /home/felics/.cache/yay/gpxviewer-git/src/gpxviewer-git
Build dir: /home/felics/.cache/yay/gpxviewer-git/src/gpxviewer-git/builddir
Build type: native build
Project name: gpx-viewer
Project version: 0.5.0
C compiler for the host machine: ccache cc (gcc 13.1.1 "cc (GCC) 13.1.1 20230714")
C linker for the host machine: cc ld.bfd 2.40.0
Vala compiler for the host machine: valac (valac 0.56.9)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Configuring config.h using configuration
Found pkg-config: /usr/bin/pkg-config (1.8.1)
Program glib-compile-resources found: YES (/usr/bin/glib-compile-resources)
Found pkg-config: /usr/bin/pkg-config (1.8.1)
Run-time dependency gmodule-2.0 found: YES 2.76.4
Run-time dependency gtk+-3.0 found: YES 3.24.38
Run-time dependency libxml-2.0 found: YES 2.11.4
Run-time dependency gdl-3.0 found: YES 3.40.0
Run-time dependency champlain-0.12 found: YES 0.12.21
Run-time dependency champlain-gtk-0.12 found: YES 0.12.21
Run-time dependency clutter-gtk-1.0 found: YES 1.8.4
Library m found: YES
Program msgfmt found: YES (/usr/bin/msgfmt)
Program msginit found: YES (/usr/bin/msginit)
Program msgmerge found: YES (/usr/bin/msgmerge)
Program xgettext found: YES (/usr/bin/xgettext)
Build targets in project: 19

gpx-viewer 0.5.0

  User defined options
    buildtype: plain
    prefix   : /usr

Found ninja-1.11.1 at /usr/bin/ninja
WARNING: Running the setup command as `meson [options]` instead of `meson setup [options]` is ambiguous and deprecated.                                      
ninja: Entering directory `builddir'
[15/33] Compiling Vala source ../libgpx/gpx-parser-fit.vala ../libgpx/gpx-con....vala ../libgpx/gpx-viewer-map-view.vala ../libgpx/gpx-viewer-path-layer.vala
FAILED: libgpx/libgpx.a.p/gpx-parser-fit.c libgpx/libgpx.a.p/gpx-convert.c libgpx/libgpx.a.p/gpx-file-base.c libgpx/libgpx.a.p/gpx-graph.c libgpx/libgpx.a.p/gpx-parser-xml.c libgpx/libgpx.a.p/gpx-parser-json.c libgpx/libgpx.a.p/gpx-point.c libgpx/libgpx.a.p/gpx-playback.c libgpx/libgpx.a.p/gpx-settings.c libgpx/libgpx.a.p/gpx-track.c libgpx/libgpx.a.p/gpx-track-treemodel.c libgpx/libgpx.a.p/gpx-viewer-map-view.c libgpx/libgpx.a.p/gpx-viewer-path-layer.c libgpx/gpx.h libgpx/gpx.vapi 
valac -C --vapidir /home/[...]/.cache/yay/gpxviewer-git/src/gpxviewer-git/vapi --pkg config --use-header --disable-assert --pkg clutter-gtk-1.0 --pkg champlain-gtk-0.12 --pkg champlain-0.12 --pkg gdl-3.0 --pkg libxml-2.0 --pkg gtk+-3.0 --pkg gmodule-2.0 --color=always --directory libgpx/libgpx.a.p --basedir ../libgpx --library gpx --header libgpx/gpx.h --vapi ../gpx.vapi ../libgpx/gpx-parser-fit.vala ../libgpx/gpx-convert.vala ../libgpx/gpx-file-base.vala ../libgpx/gpx-graph.vala ../libgpx/gpx-parser-xml.vala ../libgpx/gpx-parser-json.vala ../libgpx/gpx-point.vala ../libgpx/gpx-playback.vala ../libgpx/gpx-settings.vala ../libgpx/gpx-track.vala ../libgpx/gpx-track-treemodel.vala ../libgpx/gpx-viewer-map-view.vala ../libgpx/gpx-viewer-path-layer.vala
Command-line option `--use-header` is deprecated and will be ignored
../libgpx/gpx-parser-xml.vala:86.13-86.23: warning: `null' incompatible with return type `Gpx.Point'
   86 |             return null;
      |             ^~~~~~~~~~~ 
../libgpx/gpx-graph.vala:354.10-354.19: warning: `Gtk.Menu.popup' has been deprecated since 3.22
  354 | 					    menu.popup (null, null, null, event.button, event.time);
      | 					    ^~~~~~~~~~                                              
../libgpx/gpx-graph.vala:678.14-678.29: warning: `Gdk.cairo_create' has been deprecated since 3.22
  678 | 			var ctx = Gdk.cairo_create(win.get_window());
      | 			          ^~~~~~~~~~~~~~~~                   
../libgpx/gpx-viewer-path-layer.vala:111.3-111.33: warning: `Clutter.CairoTexture' has been deprecated since 1.12
  111 | 		Clutter.CairoTexture path_actor;
      | 		^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
../libgpx/gpx-viewer-path-layer.vala:132.4-132.22: warning: `Clutter.Actor.paint' has been deprecated since 1.12
  132 | 			content_group.paint ();
      | 			^~~~~~~~~~~~~~~~~~~    
../libgpx/gpx-viewer-path-layer.vala:138.4-138.22: warning: `Clutter.Actor.paint' has been deprecated since 1.12
  138 | 			content_group.paint ();
      | 			^~~~~~~~~~~~~~~~~~~    
../libgpx/gpx-viewer-path-layer.vala:259.3-259.31: error: overriding method `Gpx.Viewer.PathLayer.set_view' is incompatible with base method `void Champlain.Layer.set_view (Champlain.View?)': incompatible type of parameter 1.
  259 | 		public override void set_view (Champlain.View view)
      | 		^~~~~~~~~~~~~~~~~~~~~~~~~~~~~                      
../libgpx/gpx-viewer-path-layer.vala:80.3-80.21: warning: Gpx.Viewer.PathLayer.visible hides inherited property `Clutter.Actor.visible'. Use the `new' keyword if hiding was intentional
   80 | 		public bool visible {
      | 		^~~~~~~~~~~~~~~~~~~  
../libgpx/gpx-viewer-path-layer.vala:208.13-208.29: warning: `Clutter.CairoTexture.create' has been deprecated since 1.8
  208 | 			var cr = path_actor.create ();
      | 			         ^~~~~~~~~~~~~~~~~    
../libgpx/gpx-viewer-path-layer.vala:176.5-176.26: warning: `Clutter.Actor.unparent' has been deprecated since 1.10
  176 | 				content_group.unparent ();
      | 				^~~~~~~~~~~~~~~~~~~~~~    
../libgpx/gpx-viewer-path-layer.vala:122.20-122.39: warning: `Clutter.Group.new' has been deprecated since 1.10
  122 | 			content_group = new Clutter.Group ();
      | 			                ^~~~~~~~~~~~~~~~~~~~ 
../libgpx/gpx-viewer-path-layer.vala:123.4-123.27: warning: `Clutter.Actor.set_parent' has been deprecated since 1.10
  123 | 			content_group.set_parent (this);
      | 			^~~~~~~~~~~~~~~~~~~~~~~~        
../libgpx/gpx-viewer-path-layer.vala:124.21-124.40: warning: `Clutter.CairoTexture' has been deprecated since 1.12
  124 | 			path_actor = new Clutter.CairoTexture (256, 256);
      | 			                 ^~~~~~~~~~~~~~~~~~~~            
../libgpx/gpx-viewer-path-layer.vala:125.4-125.26: warning: `Clutter.Container.add_actor' has been deprecated since 1.10
  125 | 			content_group.add_actor (path_actor);
      | 			^~~~~~~~~~~~~~~~~~~~~~~              
Compilation failed: 1 error(s), 13 warning(s)
ninja: build stopped: subcommand failed.
```

Note that there is also a warning  

    ``Command-line option `--use-header` is deprecated and will be ignored``.

Tools information:  

* `valac --version`: `Vala 0.56.9`
* `ninja --version`: `1.11.1`
* `meson --version`: `1.2.0`
* `gcc --version | head -n1`: `gcc (GCC) 13.1.1 20230714`

--- a/libgpx/gpx-viewer-path-layer.vala
+++ b/libgpx/gpx-viewer-path-layer.vala
@@ -256,7 +256,11 @@ namespace Gpx.Viewer
 		ulong view_relocate_handler_id = 0U;
 		ulong view_latitude_handler_id = 0U;
 
+#if CHAMPLAIN_0_12_21
+		public override void set_view (Champlain.View? view)
+#else
 		public override void set_view (Champlain.View view)
+#endif
 		{
 			if (this.view != null) {
 				this.view.disconnect (view_relocate_handler_id);

--- a/meson.build
+++ b/meson.build
@@ -47,6 +47,10 @@ gpx_deps = [
 	meson.get_compiler('c').find_library('m'),
 ]
 
+if dependency ('champlain-0.12', version: '>=0.12.21', required: false).found()
+  add_project_arguments('-D', 'CHAMPLAIN_0_12_21', language: 'vala')
+endif
+
 subdir('libgpx')
 subdir('src')
 

--- a/nl.sarine.gpx-viewer.json
+++ b/nl.sarine.gpx-viewer.json
@@ -79,8 +79,8 @@
             "sources": [
                 {
                     "type": "archive",
-                    "url": "https://download.gnome.org/sources/libchamplain/0.12/libchamplain-0.12.20.tar.xz",
-                    "sha256": "0232b4bfcd130a1c5bda7b6aec266bf2d06e701e8093df1886f1e26bc1ba3066",
+                    "url": "https://download.gnome.org/sources/libchamplain/0.12/libchamplain-0.12.21.tar.xz",
+                    "sha256": "a915cd172a0c52944c5579fcb4683f8a878c571bf5e928254b5dafefc727e5a7",
                     "x-checker-data": {
                         "type": "gnome",
                         "name": "libchamplain"

